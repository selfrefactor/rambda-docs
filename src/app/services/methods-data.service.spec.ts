import {TestBed} from '@angular/core/testing'
import {MethodsDataService} from './methods-data.service'

describe('MethodsDataService', () => {
  let service: MethodsDataService

  beforeEach(() => {
    TestBed.configureTestingModule({})
    service = TestBed.inject(MethodsDataService)
  })

  it('should be created', () => {
    expect(service.getAllKeys()).toMatchSnapshot()
  })
  it('getActiveCategoryIndexes', () => {
    const prop = 'add'
    const a = service.getActiveCategoryIndexes({
      prop, 
      methodCategories: service.getMethod(prop).categories,
      currentFilter:'All'
    })
    console.log({a})
  })
})
