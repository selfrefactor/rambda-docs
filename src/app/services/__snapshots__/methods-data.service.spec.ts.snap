// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MethodsDataService should be created 1`] = `
Array [
  "add",
  "adjust",
  "all",
  "allPass",
  "always",
  "and",
  "or",
  "any",
  "anyPass",
  "append",
  "applySpec",
  "assoc",
  "assocPath",
  "both",
  "chain",
  "clamp",
  "clone",
  "complement",
  "compose",
  "concat",
  "cond",
  "converge",
  "curry",
  "curryN",
  "dec",
  "defaultTo",
  "difference",
  "dissoc",
  "divide",
  "drop",
  "dropLast",
  "either",
  "endsWith",
  "equals",
  "F",
  "filter",
  "find",
  "findIndex",
  "findLast",
  "findLastIndex",
  "flatten",
  "flip",
  "forEach",
  "fromPairs",
  "groupBy",
  "groupWith",
  "has",
  "hasPath",
  "head",
  "identical",
  "identity",
  "ifElse",
  "inc",
  "includes",
  "indexBy",
  "indexOf",
  "init",
  "intersection",
  "intersperse",
  "is",
  "isEmpty",
  "isNil",
  "join",
  "keys",
  "last",
  "lastIndexOf",
  "length",
  "lens",
  "lensIndex",
  "lensPath",
  "lensProp",
  "over",
  "set",
  "view",
  "map",
  "match",
  "mathMod",
  "max",
  "maxBy",
  "mean",
  "median",
  "merge",
  "mergeAll",
  "mergeDeepRight",
  "mergeLeft",
  "min",
  "minBy",
  "modulo",
  "move",
  "multiply",
  "negate",
  "none",
  "not",
  "nth",
  "once",
  "omit",
  "of",
  "partial",
  "partition",
  "path",
  "pathEq",
  "paths",
  "pathOr",
  "pick",
  "pickAll",
  "pipe",
  "pluck",
  "prepend",
  "product",
  "prop",
  "propEq",
  "propIs",
  "propOr",
  "range",
  "reduce",
  "reject",
  "repeat",
  "replace",
  "reverse",
  "slice",
  "sort",
  "sortBy",
  "split",
  "splitEvery",
  "startsWith",
  "subtract",
  "sum",
  "symmetricDifference",
  "T",
  "tail",
  "take",
  "takeLast",
  "tap",
  "test",
  "times",
  "toLower",
  "toUpper",
  "toPairs",
  "toString",
  "transpose",
  "trim",
  "tryCatch",
  "type",
  "union",
  "uniq",
  "uniqWith",
  "unless",
  "update",
  "values",
  "when",
  "where",
  "whereEq",
  "without",
  "xor",
  "zip",
  "zipObj",
  "props",
  "zipWith",
  "splitAt",
  "splitWhen",
  "takeLastWhile",
  "evolve",
  "dropLastWhile",
  "dropRepeats",
  "dropRepeatsWith",
  "dropWhile",
  "takeWhile",
  "eqProps",
]
`;
